reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -75,16 +75,17 @@ static void parse(Class<?> moduleClass, @Nullable Object instance, Scope.Builder
             }
           }
         } else {
-          if (!Modifier.isStatic(method.getModifiers()) && instance == null) {
-            // Try to just-in-time create an instance of the module using a default constructor.
-            instance = maybeInstantiate(moduleClass);
-            if (instance == null) {
-              throw new IllegalStateException(moduleClass.getCanonicalName() + " must be set");
-            }
-          }
-
           if (method.getAnnotation(Provides.class) != null) {
             ensureNotPrivate(method);
+            if (!Modifier.isStatic(method.getModifiers()) && instance == null) {
+              ensureNotAbstract(moduleClass);
```

Comment:
added extra condition to catch bad forms as seen in `ModuleAbstractInstanceProvidesMethod` and `ModuleInterfaceDefaultProvidesMethod`.